src/clr-angular/data/tree-view/models/tree-node.model.spec.ts

```diff
@@ -132,5 +132,120 @@ export default function(): void {
+        let nbChanges = 0;
+        child.disabled.subscribe(_ => nbChanges++);
+        // BehaviorSubject sends us the first state on subscription
+        expect(nbChanges).toBe(1);
+        child.setDisabled(false, false, false);
+        expect(nbChanges).toBe(1);
+        child.setDisabled(true, false, false);
+        expect(nbChanges).toBe(2);
+      });
+
+      it('can propagate the disabled state to parents', function() {
+        // Disabling all children of a node will make it disabled
+        child.children.forEach(c => c.setDisabled(true, true, false));
+        expect(child.disabled.value).toBe(true);
+        // Once the parent is disabled, a child cannot be enabled
+        child.children[0].setDisabled(false, false, false);
+        expect(child.children[1].disabled.value).toBe(true);
+      });
+
+      it('can propagate the disabled state to children', function() {
+        // Disabling a parent disables all the children down the tree
+        root.setDisabled(true, false, true);
+        [...root.children, ...child.children].forEach(n => expect(n.disabled.value).toBe(true));
+        // Enabling a parent enables all the children down the tree
+        root.setDisabled(false, false, true);
+        [...root.children, ...child.children].forEach(n => expect(n.disabled.value).toBe(false));
+        // In the case of dynamically changing parent/children, the local condition should be given
+        // higher priority than the parent policy.
+        child.setDisabled(true, false, true);
+        [...child.children].forEach(n => expect(n.disabled.value).toBe(true));
+        // local condition
+        child.children[1].isDisabledInputSetToTrue = true;
+        child.setDisabled(false, false, true);
+        expect(child.children[0].disabled.value).toBe(false);
+        expect(child.children[1].disabled.value).toBe(true);
+      });
+
+      it('completes the disabled Observable on destroy', function() {
+        let complete = false;
+        root.disabled.subscribe({ complete: () => (complete = true) });
+        root.destroy();
+        expect(complete).toBeTrue();
+      });
+
+      describe('Selection', function() {
+        it('does not toggle the selection of disabled nodes and only toggles the enabled nodes', function() {
+          root.children[1].setDisabled(true, false, false);
+          expect(root.children[0].selected.value).toBe(ClrSelectedState.UNSELECTED);
+          expect(root.children[1].selected.value).toBe(ClrSelectedState.UNSELECTED);
+          root.toggleSelection(true);
+          expect(root.children[0].selected.value).toBe(ClrSelectedState.SELECTED);
+          expect(root.children[1].selected.value).toBe(ClrSelectedState.UNSELECTED);
+        });
+
+        describe('Indeterminate parent', function() {
+          beforeEach(function() {
+            // Unselected and disabeld child
+            expect(root.children[1].selected.value).toBe(ClrSelectedState.UNSELECTED);
```

Comment:
I don't understand why you're using an `expect` in a `beforeEach`. You should only be doing setup work here.